builder/main.go

```diff
@@ -125,10 +130,14 @@ func (c *BuildCommand) Run(args []string) int {
 		err = writeJSON(config, productGrammarFile)
 		if err != nil {
 			c.Ui.Error(fmt.Sprintf("Error writing grammar file: %v", err))
+			result = multierror.Append(result, err)
 		}
 	}
-
-	return 0
+	if result.ErrorOrNil() != nil {
+		return 1
+	} else {
+		return 0
+	}
```

Comment:
```suggestion
	if err := result.ErrorOrNil(); err != nil {
		c.Ui.Error(err.Error())
		return 1
	}
	return 0
```
Just so we can also present the error messages. 😉

Reply:
Present them twice? There are still `c.Ui.Error` lines inside the for loop

Reply:
Ooops, my bad again - I didn't read the updated code carefuly enough to realize you were printing them out straight away.

![homer-fades-away](https://user-images.githubusercontent.com/287584/159749008-1478c351-54a1-410b-9458-59888119383e.gif)

Merge away! 😄